sisl/io/sile.py

```diff
@@ -562,68 +563,90 @@ def sile_read_multiple(start=0, stop=1000000000, step=None, all=False,
     all : bool, optional
        read all entries (takes precedence over start/step)
     skip_call : function, optional
-       read method without actual data processing
+       read method without actual data processing.
+       The skip call must something different from ``None`` if it succeeds.
     pre_call : function, optional
-       function to be applied before each read call
+       function to be applied before each read call.
+       Does *not* apply to the `skip_call`.
     post_call : function, optional
        function to be applied after each read call
+       Does *not* apply to the `skip_call`.
+    postprocess : function, optional
+       function to be applied on the returned data before it gets returned.
 
     Returns
     -------
     single entry or list from multiple reads
     """
     def decorator(func):
+        # ensure we can access the callables
+        nonlocal skip_call, pre_call, post_call, postprocess
+
+        if pre_call is None:
+            def pre_call(): pass
+
+        if post_call is None:
+            def post_call(): pass
+
+        if postprocess is None:
+            def postprocess(x): return x
+
+        def wrap_func(*args, **kwargs):
+            pre_call()
+            r = func(*args, **kwargs)
+            post_call()
+            return r
+
+        # if skip_call is not set, we'll simply use func.
+        # Only in this case will pre and post be called
+        # as well.
+        # they should be equivalent but skip can be used to
+        # skip some processing of the read data.
+        if skip_call is None:
+            skip_call = wrap_func
```

Comment:
I came to think that a `skip_call` should not do any of the `pre_call` and `post_call` since then it can be used in larger chunks of files. Say complicated files that doesn't really fit regular data files.

It may be weird for users supplying a skip call to be wrapped with the pre and post if they are not needed. It is simpler to manually put them in if needed.